In [1]:
#a code block that deals with concentration and drug names
#we do not use this block for now.

import pandas as pd
f1 = pd.read_csv('B.csv')
f2 = pd.read_csv('C.csv')
f3=pd.read_csv('D.csv')
f4=pd.read_csv('E.csv')
f5=pd.read_csv('A.csv')
file = [f1,f2,f3,f5]
train = pd.concat(file,axis=1)
train.to_csv("train.csv", index=0, sep=',')


In [2]:
#do some further processing with "A" because there are some data error in it
def remove_extradots(string):
    counter=0
    lst=[]
    for i in range(len(string)):
        if string[i]=='.' and counter<1 :
            lst.append(string[i])
            counter+=1
        elif string[i]!='.':
            lst.append(string[i])
    res="".join(lst)
    return res



import csv
res_list=[]
print(remove_extradots("-0.29444.2"))
with open('A.csv') as file2:
    reader=csv.reader(file2)

    for row in reader:
        data=[]
        for i in range(len(row)):
            
            data.append(remove_extradots(row[i]))
        res_list.append(data)
    


with open('modified_A.csv', 'w') as file3:
    writer = csv.writer(file3)
    writer.writerows(res_list)
file3.close()
                


-0.294442


In [3]:
#test file A
import pandas as pd
df_A=pd.read_csv("A.csv")
df_A.shape
#df_A2=pd.read_csv("A_modified.csv")
#df_A2.shape


(767, 239)

In [4]:
#a block that is used to concate the drug names(drug combinations and related concentration)

import csv
res_list=[]
with open('train.csv', 'r') as file:
    reader = csv.reader(file)
    
    for row in reader:
        data=[]
        data.append(row[0]+'+'+row[1]+"+"+row[2])
        res_list.append(data)

        
        
with open('concatenated.csv', 'w') as file2:
    writer = csv.writer(file2)
    writer.writerows(res_list)

In [5]:
#a block that is used to concate concentration and ks numbers.
import pandas as pd
f1 = pd.read_csv('concatenated.csv')
#f2 = pd.read_csv('C.csv')
f3=pd.read_csv('D.csv')
f4=pd.read_csv('E.csv')
f5=pd.read_csv('modified_A.csv')
file = [f1,f5]
train = pd.concat(file,axis=1)
train.to_csv("train3.csv", index=0, sep=',')

In [6]:
#a block that sets the fields for the drug data
import csv
fields=['drug_name','default']
for i in range(1,239):
    fields.append('ks'+str(i))
rows=[]
with open('train3.csv', 'r') as file:
    reader = csv.reader(file)
    
    for row in reader:
        rows.append(row)

        
        
with open('drug_withfields2.csv', 'w') as file2:
    writer = csv.writer(file2)
    writer.writerow(fields)
    writer.writerows(rows)

In [7]:
#do some further processing with "drug_withfields" because there are still some data error in it
def remove_extradots(string):
    counter=0
    lst=[]
    for i in range(len(string)):
        if string[i]=='.' and counter<1 :
            lst.append(string[i])
            counter+=1
        elif string[i]!='.':
            lst.append(string[i])
    res="".join(lst)
    return res



import csv
res_list=[]
print(remove_extradots("-0.29444.2"))
with open('drug_withfields2.csv') as file2:
    reader=csv.reader(file2)

    for row in reader:
        data=[]
        for i in range(len(row)):
            
            data.append(remove_extradots(row[i]))
        res_list.append(data)
    


with open('drug_with_fields.csv', 'w') as file3:
    writer = csv.writer(file3)
    writer.writerows(res_list)
file3.close()
                

-0.294442


In [26]:
df=pd.read_csv('drug_with_fields.csv')
#one type first type. dna
df11=df.loc[(df['drug_name'] =='Etoposide+Gemcitabine+1') ]
df12=df.loc[(df['drug_name'] =='Etoposide+Gemcitabine+2') ]
df13=df.loc[(df['drug_name'] =='Etoposide+Gemcitabine+3') ]
df14=df.loc[(df['drug_name'] =='Etoposide+Gemcitabine+4') ]
df15=df.loc[(df['drug_name'] =='Etoposide+Gemcitabine+5') ]
df16=df.loc[(df['drug_name'] =='Etoposide+Gemcitabine+6') ]
df21=df.loc[(df['drug_name'] =='Gemcitabine+Etoposide+1') ]
df22=df.loc[(df['drug_name'] =='Gemcitabine+Etoposide+2') ]
df23=df.loc[(df['drug_name'] =='Gemcitabine+Etoposide+3') ]
df24=df.loc[(df['drug_name'] =='Gemcitabine+Etoposide+4') ]
df25=df.loc[(df['drug_name'] =='Gemcitabine+Etoposide+5') ]
df26=df.loc[(df['drug_name'] =='Gemcitabine+Etoposide+6') ]
df31=df.loc[(df['drug_name'] =='Etoposide+Etoposide+1') ]
df32=df.loc[(df['drug_name'] =='Etoposide+Etoposide+2') ]
df33=df.loc[(df['drug_name'] =='Etoposide+Etoposide+3') ]
df34=df.loc[(df['drug_name'] =='Etoposide+Etoposide+4') ]
df35=df.loc[(df['drug_name'] =='Etoposide+Etoposide+5') ]
df36=df.loc[(df['drug_name'] =='Etoposide+Etoposide+6') ]
df41=df.loc[(df['drug_name'] =='Gemcitabine+Gemcitabine+1') ]
df42=df.loc[(df['drug_name'] =='Gemcitabine+Gemcitabine+2') ]
df43=df.loc[(df['drug_name'] =='Gemcitabine+Gemcitabine+3') ]
df44=df.loc[(df['drug_name'] =='Gemcitabine+Gemcitabine+4') ]
df45=df.loc[(df['drug_name'] =='Gemcitabine+Gemcitabine+5') ]
df46=df.loc[(df['drug_name'] =='Gemcitabine+Gemcitabine+6') ]
frames = [df11, df12,df13,df14,df15,df16,df21,df22,df23,df24,df25,df26,df31,df32,df33,df34,df35,df36,df41,df42,df43,df44,df45,df46]
dna = pd.concat(frames)
dna


drug_name  default       ks1       ks2       ks3  \
12    Etoposide+Gemcitabine+1        0 -0.825000 -0.463430 -0.623680   
14    Etoposide+Gemcitabine+2        0 -0.850830 -0.166270  0.283220   
16    Etoposide+Gemcitabine+3        0 -0.690000  0.233580 -0.093266   
18    Etoposide+Gemcitabine+4        0 -0.682500  0.227200  0.109890   
20    Etoposide+Gemcitabine+5        0 -0.153740 -0.136810 -0.168360   
22    Etoposide+Gemcitabine+6        0  0.023568 -0.042826  0.057062   
72    Gemcitabine+Etoposide+1        0 -0.825000 -0.537260 -0.763460   
74    Gemcitabine+Etoposide+2        0 -0.901670 -0.396500 -0.427270   
76    Gemcitabine+Etoposide+3        0 -0.864170 -0.188210  0.287800   
78    Gemcitabine+Etoposide+4        0  0.070186 -0.086002 -0.037225   
80    Gemcitabine+Etoposide+5        0  0.090969 -0.095617  0.039580   
82    Gemcitabine+Etoposide+6        0  0.102160 -0.140670 -0.078014   
0       Etoposide+Etoposide+1        0 -0.915000 -0.309170 -0.266390   
2       Etoposide+Etoposide+2        0 -0.827500 -0.185050  0.225330   
4       Etoposide+Etoposide+3        0 -0.590000  0.163430 -0.071733   
6       Etoposide+Etoposide+4        0  0.055863 -0.089835 -0.060518   
8       Etoposide+Etoposide+5        0  0.072606 -0.050656  0.029489   
10      Etoposide+Etoposide+6        0  0.051316 -0.086824 -0.048019   
84  Gemcitabine+Gemcitabine+1        0 -0.737500 -0.573210 -0.803810   
86  Gemcitabine+Gemcitabine+2        0 -0.770830 -0.508760 -0.646980   
88  Gemcitabine+Gemcitabine+3        0 -0.850830 -0.219490 -0.226760   
90  Gemcitabine+Gemcitabine+4        0 -0.695830  0.267190  0.171970   
92  Gemcitabine+Gemcitabine+5        0 -0.103980 -0.160460 -0.235150   
94  Gemcitabine+Gemcitabine+6        0  0.080430 -0.081518  0.046713   

         ks4       ks5       ks6       ks7       ks8  ...     ks229     ks230  \
12 -0.811710 -0.887700 -0.625570 -0.599290 -0.572840  ... -0.185950  0.197050   
14 -0.518290 -0.644780 -0.131510  0.334720  0.418330  ...  0.173700  0.161250   
16  0.163760  0.195020 -0.093485 -0.094498 -0.089402  ...  0.484000  0.292200   
18  0.123580  0.155430  0.110520  0.096807  0.088164  ...  0.407050  0.313020   
20 -0.151440 -0.222490 -0.169770 -0.157740 -0.151010  ...  0.036541  0.060248   
22 -0.089556  0.059031  0.059376  0.054753  0.044795  ... -0.080531  0.050176   
72 -0.951370 -0.966760 -0.781250 -0.743160 -0.715600  ... -0.252210  0.148460   
74 -0.722000 -0.814450 -0.443990 -0.394930 -0.363790  ... -0.143550  0.188880   
76 -0.461490 -0.612800 -0.174290  0.321490  0.384460  ...  0.164050  0.166790   
78 -0.093018 -0.043573 -0.044761 -0.034446  0.030112  ... -0.136540  0.031322   
80 -0.026449 -0.053456  0.016219  0.063441  0.065181  ... -0.161020  0.042661   
82 -0.119010 -0.125790 -0.079357 -0.064932 -0.062010  ... -0.164220  0.027195   
0  -0.595000 -0.719440 -0.304440 -0.235280  0.280280  ...  0.096111  0.273331   
2  -0.530960 -0.628240 -0.191880  0.273000  0.321280  ...  0.172830  0.209750   
4   0.036633  0.147570 -0.074267 -0.072833 -0.063000  ...  0.366530  0.210270   
6  -0.058003 -0.082102 -0.062598 -0.054715 -0.045668  ... -0.148370  0.056149   
8  -0.074577 -0.041196 -0.023138  0.033754  0.037671  ... -0.111160  0.032801   
10 -0.051763 -0.092343 -0.050676 -0.042838 -0.037814  ... -0.111460  0.036606   
84 -0.929760 -0.971900 -0.815710 -0.786190 -0.757140  ... -0.314880  0.123330   
86 -0.879480 -0.954010 -0.670320 -0.623650 -0.594750  ... -0.283330  0.169460   
88 -0.482250 -0.595680 -0.243110  0.272250  0.334990  ...  0.136960  0.184610   
90  0.111910  0.149880  0.163190  0.147090  0.149650  ...  0.424790  0.281040   
92 -0.199160 -0.321510 -0.228840 -0.211470 -0.200060  ... -0.057266  0.065418   
94 -0.044981 -0.035029  0.034172  0.057867  0.063030  ... -0.138880  0.056514   

       ks231     ks232     ks233     ks234     ks235     ks236     ks237  \
12 -0.246260 -0.134130 -0.121170  0.197050 -0.292570 -0.349690 -0.322820   
14 -0.391330  0.255540  0.227910  0.16125

In [27]:

#one type first type. mt
df11=df.loc[(df['drug_name'] =='Paclitaxel+Vinblastine+1') ]
df12=df.loc[(df['drug_name'] =='Paclitaxel+Vinblastine+2') ]
df13=df.loc[(df['drug_name'] =='Paclitaxel+Vinblastine+3') ]
df14=df.loc[(df['drug_name'] =='Paclitaxel+Vinblastine+4') ]
df15=df.loc[(df['drug_name'] =='Paclitaxel+Vinblastine+5') ]
df16=df.loc[(df['drug_name'] =='Paclitaxel+Vinblastine+6') ]
df21=df.loc[(df['drug_name'] =='Vinblastine+Paclitaxel+1') ]
df22=df.loc[(df['drug_name'] =='Vinblastine+Paclitaxel+2') ]
df23=df.loc[(df['drug_name'] =='Vinblastine+Paclitaxel+3') ]
df24=df.loc[(df['drug_name'] =='Vinblastine+Paclitaxel+4') ]
df25=df.loc[(df['drug_name'] =='Vinblastine+Paclitaxel+5') ]
df26=df.loc[(df['drug_name'] =='Vinblastine+Paclitaxel+6') ]
df31=df.loc[(df['drug_name'] =='Paclitaxel+Paclitaxel+1') ]
df32=df.loc[(df['drug_name'] =='Paclitaxel+Paclitaxel+2') ]
df33=df.loc[(df['drug_name'] =='Paclitaxel+Paclitaxel+3') ]
df34=df.loc[(df['drug_name'] =='Paclitaxel+Paclitaxel+4') ]
df35=df.loc[(df['drug_name'] =='Paclitaxel+Paclitaxel+5') ]
df36=df.loc[(df['drug_name'] =='Paclitaxel+Paclitaxel+6') ]
df41=df.loc[(df['drug_name'] =='Vinblastine+Vinblastine+1') ]
df42=df.loc[(df['drug_name'] =='Vinblastine+Vinblastine+2') ]
df43=df.loc[(df['drug_name'] =='Vinblastine+Vinblastine+3') ]
df44=df.loc[(df['drug_name'] =='Vinblastine+Vinblastine+4') ]
df45=df.loc[(df['drug_name'] =='Vinblastine+Vinblastine+5') ]
df46=df.loc[(df['drug_name'] =='Vinblastine+Vinblastine+6') ]
frames2 = [df11, df12,df13,df14,df15,df16,df21,df22,df23,df24,df25,df26,df31,df32,df33,df34,df35,df36,df41,df42,df43,df44,df45,df46]
mt= pd.concat(frames)
mt



drug_name  default       ks1       ks2       ks3  \
12    Etoposide+Gemcitabine+1        0 -0.825000 -0.463430 -0.623680   
14    Etoposide+Gemcitabine+2        0 -0.850830 -0.166270  0.283220   
16    Etoposide+Gemcitabine+3        0 -0.690000  0.233580 -0.093266   
18    Etoposide+Gemcitabine+4        0 -0.682500  0.227200  0.109890   
20    Etoposide+Gemcitabine+5        0 -0.153740 -0.136810 -0.168360   
22    Etoposide+Gemcitabine+6        0  0.023568 -0.042826  0.057062   
72    Gemcitabine+Etoposide+1        0 -0.825000 -0.537260 -0.763460   
74    Gemcitabine+Etoposide+2        0 -0.901670 -0.396500 -0.427270   
76    Gemcitabine+Etoposide+3        0 -0.864170 -0.188210  0.287800   
78    Gemcitabine+Etoposide+4        0  0.070186 -0.086002 -0.037225   
80    Gemcitabine+Etoposide+5        0  0.090969 -0.095617  0.039580   
82    Gemcitabine+Etoposide+6        0  0.102160 -0.140670 -0.078014   
0       Etoposide+Etoposide+1        0 -0.915000 -0.309170 -0.266390   
2       Etoposide+Etoposide+2        0 -0.827500 -0.185050  0.225330   
4       Etoposide+Etoposide+3        0 -0.590000  0.163430 -0.071733   
6       Etoposide+Etoposide+4        0  0.055863 -0.089835 -0.060518   
8       Etoposide+Etoposide+5        0  0.072606 -0.050656  0.029489   
10      Etoposide+Etoposide+6        0  0.051316 -0.086824 -0.048019   
84  Gemcitabine+Gemcitabine+1        0 -0.737500 -0.573210 -0.803810   
86  Gemcitabine+Gemcitabine+2        0 -0.770830 -0.508760 -0.646980   
88  Gemcitabine+Gemcitabine+3        0 -0.850830 -0.219490 -0.226760   
90  Gemcitabine+Gemcitabine+4        0 -0.695830  0.267190  0.171970   
92  Gemcitabine+Gemcitabine+5        0 -0.103980 -0.160460 -0.235150   
94  Gemcitabine+Gemcitabine+6        0  0.080430 -0.081518  0.046713   

         ks4       ks5       ks6       ks7       ks8  ...     ks229     ks230  \
12 -0.811710 -0.887700 -0.625570 -0.599290 -0.572840  ... -0.185950  0.197050   
14 -0.518290 -0.644780 -0.131510  0.334720  0.418330  ...  0.173700  0.161250   
16  0.163760  0.195020 -0.093485 -0.094498 -0.089402  ...  0.484000  0.292200   
18  0.123580  0.155430  0.110520  0.096807  0.088164  ...  0.407050  0.313020   
20 -0.151440 -0.222490 -0.169770 -0.157740 -0.151010  ...  0.036541  0.060248   
22 -0.089556  0.059031  0.059376  0.054753  0.044795  ... -0.080531  0.050176   
72 -0.951370 -0.966760 -0.781250 -0.743160 -0.715600  ... -0.252210  0.148460   
74 -0.722000 -0.814450 -0.443990 -0.394930 -0.363790  ... -0.143550  0.188880   
76 -0.461490 -0.612800 -0.174290  0.321490  0.384460  ...  0.164050  0.166790   
78 -0.093018 -0.043573 -0.044761 -0.034446  0.030112  ... -0.136540  0.031322   
80 -0.026449 -0.053456  0.016219  0.063441  0.065181  ... -0.161020  0.042661   
82 -0.119010 -0.125790 -0.079357 -0.064932 -0.062010  ... -0.164220  0.027195   
0  -0.595000 -0.719440 -0.304440 -0.235280  0.280280  ...  0.096111  0.273331   
2  -0.530960 -0.628240 -0.191880  0.273000  0.321280  ...  0.172830  0.209750   
4   0.036633  0.147570 -0.074267 -0.072833 -0.063000  ...  0.366530  0.210270   
6  -0.058003 -0.082102 -0.062598 -0.054715 -0.045668  ... -0.148370  0.056149   
8  -0.074577 -0.041196 -0.023138  0.033754  0.037671  ... -0.111160  0.032801   
10 -0.051763 -0.092343 -0.050676 -0.042838 -0.037814  ... -0.111460  0.036606   
84 -0.929760 -0.971900 -0.815710 -0.786190 -0.757140  ... -0.314880  0.123330   
86 -0.879480 -0.954010 -0.670320 -0.623650 -0.594750  ... -0.283330  0.169460   
88 -0.482250 -0.595680 -0.243110  0.272250  0.334990  ...  0.136960  0.184610   
90  0.111910  0.149880  0.163190  0.147090  0.149650  ...  0.424790  0.281040   
92 -0.199160 -0.321510 -0.228840 -0.211470 -0.200060  ... -0.057266  0.065418   
94 -0.044981 -0.035029  0.034172  0.057867  0.063030  ... -0.138880  0.056514   

       ks231     ks232     ks233     ks234     ks235     ks236     ks237  \
12 -0.246260 -0.134130 -0.121170  0.197050 -0.292570 -0.349690 -0.322820   
14 -0.391330  0.255540  0.227910  0.16125

In [109]:
#Dna ： etoposide，Gemcitabine
#hsp：17aag，auy922
#mt：paclitaxel，Vinblastine
#prot："MG132"，ps341
#hdac：saha，apicidin

frames3=[]
for i in range(1,7):
    frames3.append(df.loc[(df['drug_name'] =='17AAG+AUY922+'+str(i)) ])
for j in range(1,7):
    frames3.append(df.loc[(df['drug_name'] =='AUY922+17AAG+'+str(j)) ])
for k in range(1,7):
    frames3.append(df.loc[(df['drug_name'] =='AUY922+AUY922+'+str(k)) ])
for m in range(1,7):
    frames3.append(df.loc[(df['drug_name'] =='17AAG+17AAG+'+str(m)) ])
HSP= pd.concat(frames3)
HSP


drug_name  default       ks1       ks2       ks3       ks4  \
397   17AAG+AUY922+1        0 -0.767500 -0.577600 -0.800500 -0.933350   
399   17AAG+AUY922+2        0 -0.855830 -0.333280 -0.409610 -0.662020   
401   17AAG+AUY922+3        0 -0.895830 -0.218060  0.233330 -0.531440   
403   17AAG+AUY922+4        0 -0.313330  0.078268 -0.065891  0.144930   
405   17AAG+AUY922+5        0  0.055394  0.062562  0.064600  0.111320   
407   17AAG+AUY922+6        0  0.067365  0.085337  0.117400  0.102500   
457   AUY922+17AAG+1        0 -0.652500 -0.527100 -0.758780 -0.902230   
459   AUY922+17AAG+2        0 -0.799170 -0.140630  0.179360 -0.339620   
461   AUY922+17AAG+3        0 -0.654170  0.298920 -0.103650  0.255280   
463   AUY922+17AAG+4        0 -0.657500  0.272850  0.113280  0.313410   
465   AUY922+17AAG+5        0  0.082597 -0.084254 -0.123730  0.158700   
467   AUY922+17AAG+6        0  0.091737 -0.073393 -0.076631  0.173550   
469  AUY922+AUY922+1        0 -0.930000 -0.269100 -0.213840 -0.491870   
471  AUY922+AUY922+2        0 -0.766670 -0.483180 -0.638680 -0.839050   
473  AUY922+AUY922+3        0 -0.857500 -0.262300 -0.219090 -0.526060   
475  AUY922+AUY922+4        0 -0.729170  0.285200  0.198150  0.131000   
477  AUY922+AUY922+5        0 -0.102920 -0.160560 -0.232300 -0.141410   
479  AUY922+AUY922+6        0  0.108200 -0.044401 -0.020423 -0.048169   
385    17AAG+17AAG+1        0 -0.896670 -0.225090  0.219610 -0.421540   
387    17AAG+17AAG+2        0 -0.830000 -0.158990  0.224970 -0.337900   
389    17AAG+17AAG+3        0 -0.523330  0.195240 -0.084601  0.088844   
391    17AAG+17AAG+4        0  0.126000 -0.054614  0.042599  0.029917   
393    17AAG+17AAG+5        0  0.096792  0.047683  0.031541  0.084167   
395    17AAG+17AAG+6        0  0.068132  0.050531  0.040130  0.152930   

          ks5       ks6       ks7       ks8  ...     ks229     ks230  \
397 -0.975430 -0.812160 -0.780090 -0.755510  ... -0.291430  0.144300   
399 -0.859170 -0.472990 -0.374080 -0.350350  ... -0.102560  0.147570   
401 -0.747220 -0.265600  0.282940  0.333660  ...  0.097255  0.199840   
403  0.069141 -0.063347 -0.059033 -0.057116  ...  0.222320  0.110430   
405  0.055389  0.052876  0.069045  0.068527  ... -0.048431  0.026449   
407  0.092460  0.099398  0.112610  0.118850  ... -0.050581  0.042649   
457 -0.952800 -0.787750 -0.724840 -0.700890  ... -0.234310  0.135380   
459 -0.543380 -0.159550  0.226200  0.282870  ...  0.299740  0.196710   
461 -0.103130 -0.106240 -0.097799 -0.088767  ...  0.487240  0.275650   
463  0.116830 -0.091533  0.105540  0.094172  ...  0.400790  0.246300   
465 -0.217290 -0.140290 -0.097177 -0.090041  ... -0.112220  0.042521   
467 -0.146350 -0.101760 -0.056243 -0.050155  ... -0.122390  0.029006   
469 -0.761870 -0.221860  0.274810  0.365440  ...  0.165480  0.129470   
471 -0.946340 -0.661920 -0.614170 -0.580890  ... -0.191220  0.140600   
473 -0.747420 -0.246160  0.278610  0.354950  ...  0.110380  0.205830   
475  0.116790  0.147280  0.178190  0.170730  ...  0.446890  0.311850   
477 -0.303500 -0.228620 -0.202780 -0.201210  ...  0.033686  0.066144   
479 -0.025305 -0.031268 -0.026796 -0.017418  ... -0.102590  0.047242   
385 -0.582240 -0.188330  0.266580  0.347020  ...  0.147190  0.222890   
387 -0.560110 -0.156190  0.264920  0.314830  ...  0.240640  0.227370   
389  0.168130  0.086149 -0.090917 -0.077238  ...  0.373010  0.215470   
391 -0.042958 -0.043068  0.046661  0.060640  ... -0.090386  0.022809   
393  0.057386  0.024384  0.021610  0.030651  ... -0.088961  0.026336   
395  0.026753  0.027607  0.041234  0.046726  ... -0.060887  0.050492   

        ks231     ks232     ks233     ks234     ks235     ks236     ks237  \
397  0.100320 -0.258110 -0.223520  0.144300  0.184040  0.154490 -0.225680   
399 -0.185850 -0.116840 -0.093259  0.147570 -0.379310 -0.399990 -0.343510   
401 -0.267600  0.150700  0.116470  0.199840 -0.603610 -0.589440 -0.420080   
403 -0.129760  0.271310  0.243000  0.110430 -0.247570 -0.240640 -0.122680   

In [111]:
#Dna ： etoposide，Gemcitabine
#mt：paclitaxel，Vinblastine
#hsp：17aag，auy922

#prot："MG132"，ps341
#hdac：saha，apicidin

frames4=[]
for i in range(1,7):
    frames4.append(df.loc[(df['drug_name'] =='MG132+PS341+'+str(i)) ])
for j in range(1,7):
    frames4.append(df.loc[(df['drug_name'] =='PS341+MG132+'+str(j)) ])
for k in range(1,7):
    frames4.append(df.loc[(df['drug_name'] =='MG132+MG132+'+str(k)) ])
for m in range(1,7):
    frames4.append(df.loc[(df['drug_name'] =='PS341+PS341+'+str(m)) ])
PROT= pd.concat(frames4)
PROT

drug_name  default       ks1       ks2       ks3       ks4       ks5  \
541  MG132+PS341+1        0 -0.980830  0.275420  0.463330 -0.309580  0.193750   
543  MG132+PS341+2        0 -0.560000  0.530740  0.558230  0.126370  0.610510   
545  MG132+PS341+3        0 -0.352500  0.313820  0.287250  0.204430  0.248170   
547  MG132+PS341+4        0 -0.201720  0.217710  0.269260  0.109690  0.220390   
549  MG132+PS341+5        0 -0.066823  0.069562  0.086963  0.017766  0.051397   
551  MG132+PS341+6        0  0.096792 -0.029772  0.051313 -0.073721  0.032557   
601  PS341+MG132+1        0 -0.201670 -0.780090 -0.999200 -0.999200 -1.000000   
603  PS341+MG132+2        0 -0.842500  0.222630  0.417240 -0.108640  0.102980   
605  PS341+MG132+3        0 -0.601670  0.329910  0.396860  0.094715  0.267520   
607  PS341+MG132+4        0 -0.407500  0.371080  0.421490  0.140580  0.406780   
609  PS341+MG132+5        0 -0.139100  0.040512 -0.064549 -0.025732  0.058738   
611  PS341+MG132+6        0  0.111620 -0.103320 -0.119630  0.073187 -0.159250   
529  MG132+MG132+1        0 -0.202530 -0.771890 -0.996780 -1.000000 -1.000000   
531  MG132+MG132+2        0 -0.550270  0.265850  0.320740  0.096816  0.242260   
533  MG132+MG132+3        0 -0.352500  0.237920  0.234320  0.138870  0.130440   
535  MG132+MG132+4        0 -0.298330  0.202930  0.258550  0.109530  0.181600   
537  MG132+MG132+5        0 -0.114170  0.093214  0.112740  0.039167  0.068571   
539  MG132+MG132+6        0  0.104270 -0.097685 -0.066722 -0.086166 -0.123200   
613  PS341+PS341+1        0 -0.131670 -0.798930 -0.997070 -1.000000 -1.000000   
615  PS341+PS341+2        0 -0.820830  0.415250  0.444770  0.075250  0.489880   
617  PS341+PS341+3        0 -0.531670  0.563460  0.624960  0.154780  0.461170   
619  PS341+PS341+4        0 -0.434170  0.327380  0.373860  0.137210  0.365040   
621  PS341+PS341+5        0 -0.165370 -0.032087 -0.087419 -0.060405 -0.108050   
623  PS341+PS341+6        0  0.069655 -0.053471 -0.031627  0.054596  0.025207   

          ks6       ks7       ks8  ...     ks229     ks230     ks231  \
541  0.522500  0.439170  0.423330  ...  0.390830  0.215830 -0.303750   
543  0.585780  0.518460  0.490410  ...  0.513160  0.048574 -0.140000   
545  0.253590  0.282090  0.286860  ...  0.284590  0.066781  0.042996   
547  0.265700  0.273090  0.256170  ...  0.184200  0.042661  0.048200   
549  0.099413  0.091493  0.091541  ...  0.050442  0.059708 -0.054403   
551  0.043402  0.046963  0.053984  ... -0.099532  0.023664  0.025799   
601 -0.999200 -0.997590 -0.997590  ... -0.526240 -0.080707  0.465770   
603  0.376360  0.436890  0.454610  ...  0.186540 -0.162020  0.178160   
605  0.346340  0.400990  0.426400  ...  0.390430  0.087546 -0.113340   
607  0.407920  0.396360  0.387970  ...  0.348590  0.112400 -0.065488   
609 -0.066280 -0.068681 -0.070268  ...  0.106820  0.045190 -0.053640   
611 -0.127380 -0.113350 -0.096428  ... -0.122150  0.028359  0.042466   
529 -0.997580 -0.993550 -0.991140  ... -0.532800 -0.065368  0.466160   
531  0.326730  0.314770  0.297960  ...  0.240730  0.247530 -0.102150   
533  0.184970  0.242750  0.261990  ...  0.264750  0.128420 -0.045300   
535  0.245410  0.256650  0.248360  ...  0.228270  0.095216 -0.058033   
537  0.095357  0.113330  0.117500  ...  0.088571  0.050833 -0.042976   
539 -0.065869 -0.055640 -0.047237  ... -0.113790  0.029607 -0.022547   
613 -0.998530 -0.995600 -0.992670  ... -0.553190 -0.089377  0.507810   
615  0.468310  0.403780  0.393260  ...  0.496810 -0.070117 -0.212380   
617  0.615920  0.585940  0.570700  ...  0.383420  0.054898  0.039751   
619  0.350310  0.346320  0.348640  ...  0.312020  0.089419 -0.034884   
621 -0.092221 -0.083718 -0.079547  ...  0.106480  0.037262 -0.041725   
623 -0.032786 -0.030799  0.025108  ... -0.051282 -0.023373  0.030227   

        ks232     ks233     ks234     ks235     ks236     ks237     ks238  
541  0.471670  0.406670  0.215830 -0.515420 -0.505830 -0.282500  0.353330  
543  0.603710  0.608370  0.048

In [112]:
#Dna ： etoposide，Gemcitabine
#hsp：17aag，auy922
#mt：paclitaxel，Vinblastine
#prot："MG132"，ps341
#hdac：saha，apicidin

frames5=[]
for i in range(1,7):
    frames5.append(df.loc[(df['drug_name'] =='SAHA+Apicidin+'+str(i)) ])
for j in range(1,7):
    frames5.append(df.loc[(df['drug_name'] =='Apicidin+SAHA+'+str(j)) ])
for k in range(1,7):
    frames5.append(df.loc[(df['drug_name'] =='Apicidin+Apicidin+'+str(k)) ])
for m in range(1,7):
    frames5.append(df.loc[(df['drug_name'] =='SAHA+SAHA+'+str(m)) ])
HDAC= pd.concat(frames5)
HDAC

drug_name  default       ks1       ks2       ks3       ks4  \
672      SAHA+Apicidin+1        0 -0.895830  0.214310  0.321900 -0.144170   
674      SAHA+Apicidin+2        0 -0.575830  0.546330  0.570710  0.147600   
676      SAHA+Apicidin+3        0 -0.075150  0.416830  0.464100  0.192980   
678      SAHA+Apicidin+4        0 -0.043026  0.097982  0.071053  0.119250   
680      SAHA+Apicidin+5        0 -0.054167 -0.090500 -0.122330 -0.053667   
682      SAHA+Apicidin+6        0  0.061709 -0.115710 -0.127120  0.061750   
289      Apicidin+SAHA+1        0 -0.937500  0.275810  0.458540 -0.191050   
291      Apicidin+SAHA+2        0 -0.605840  0.665340  0.722010  0.169270   
293      Apicidin+SAHA+3        0  0.048769  0.322460  0.465320  0.156850   
295      Apicidin+SAHA+4        0  0.081883 -0.031892  0.042259  0.074661   
297      Apicidin+SAHA+5        0  0.035269 -0.084705 -0.034474 -0.035295   
299      Apicidin+SAHA+6        0  0.064319 -0.110200 -0.102150  0.077367   
673  Apicidin+Apicidin+1        0 -0.935000  0.359170  0.515830 -0.144170   
675  Apicidin+Apicidin+2        0 -0.536670  0.603580  0.645670  0.184680   
677  Apicidin+Apicidin+3        0 -0.077500  0.400230  0.448270  0.188270   
679  Apicidin+Apicidin+4        0  0.057114  0.069390  0.087154  0.136990   
681  Apicidin+Apicidin+5        0  0.040290 -0.053320 -0.056615  0.078565   
683  Apicidin+Apicidin+6        0  0.075676 -0.124450 -0.200030  0.082567   
288          SAHA+SAHA+1        0 -0.954170  0.308430  0.481910 -0.112810   
290          SAHA+SAHA+2        0 -0.575830  0.614180  0.676500  0.127380   
292          SAHA+SAHA+3        0 -0.039234  0.357270  0.467940  0.189710   
294          SAHA+SAHA+4        0  0.042073  0.090531  0.140260  0.093571   
296          SAHA+SAHA+5        0 -0.037934 -0.052660 -0.028860  0.057298   
298          SAHA+SAHA+6        0 -0.025114 -0.112470 -0.111420 -0.072167   

          ks5       ks6       ks7       ks8  ...     ks229     ks230  \
672 -0.186290  0.260520  0.326450  0.312860  ...  0.315390  0.325510   
674  0.561130  0.567480  0.548690  0.526090  ...  0.486660  0.185000   
676  0.510480  0.469800  0.440490  0.421940  ...  0.139590  0.035929   
678  0.126800  0.074386  0.063026  0.062325  ...  0.048553 -0.019474   
680 -0.150670 -0.115500 -0.112170 -0.105670  ... -0.030833  0.041500   
682 -0.136240 -0.136240 -0.119760 -0.113210  ... -0.091638  0.036659   
289 -0.134840  0.456990  0.477240  0.461580  ...  0.295390  0.359740   
291  0.710350  0.724530  0.702020  0.670370  ...  0.521240  0.234950   
293  0.517990  0.464020  0.434210  0.410130  ...  0.024747  0.039193   
295  0.084931  0.043469 -0.041731 -0.051409  ... -0.120470 -0.028830   
297 -0.046115 -0.033526 -0.030013 -0.037538  ... -0.105600  0.040154   
299 -0.115610 -0.115270 -0.087476 -0.082872  ... -0.131390  0.028050   
673 -0.150000  0.440000  0.535420  0.536250  ...  0.281250  0.265000   
675  0.644680  0.659080  0.610900  0.587040  ...  0.506500  0.136740   
677  0.529200  0.453420  0.418060  0.395910  ...  0.154200  0.029927   
679  0.148540  0.092215  0.078130  0.080894  ... -0.053211  0.041809   
681 -0.027636 -0.050113 -0.049189 -0.053397  ... -0.057538  0.056320   
683 -0.220830 -0.209390 -0.180830 -0.167210  ... -0.085052  0.040278   
288 -0.147410  0.432840  0.482070  0.465560  ...  0.383730  0.275490   
290  0.676570  0.688240  0.657050  0.620200  ...  0.501140  0.155110   
292  0.540580  0.464200  0.437200  0.413260  ...  0.075193  0.030012   
294  0.183460  0.141350  0.129490  0.116210  ... -0.063889  0.024933   
296  0.026528 -0.030210 -0.030452 -0.033717  ... -0.067626  0.041535   
298 -0.133120 -0.117890 -0.103570 -0.097284  ... -0.070852 -0.022251   

        ks231     ks232     ks233     ks234     ks235     ks236     ks237  \
672 -0.404330  0.461390  0.432160  0.325510 -0.542860 -0.613430 -0.420020   
674 -0.171040  0.604080  0.597490  0.185000 -0.511500 -0.486640 -0.132980   
676  0.061585  0.155230  0.167830  0.035929  0.184770  0.

In [115]:
combi=frames+frames2+frames3+frames4+frames5
training= pd.concat(combi)
training



drug_name  default       ks1       ks2       ks3       ks4  \
12   Etoposide+Gemcitabine+1        0 -0.825000 -0.463430 -0.623680 -0.811710   
14   Etoposide+Gemcitabine+2        0 -0.850830 -0.166270  0.283220 -0.518290   
16   Etoposide+Gemcitabine+3        0 -0.690000  0.233580 -0.093266  0.163760   
18   Etoposide+Gemcitabine+4        0 -0.682500  0.227200  0.109890  0.123580   
20   Etoposide+Gemcitabine+5        0 -0.153740 -0.136810 -0.168360 -0.151440   
..                       ...      ...       ...       ...       ...       ...   
290              SAHA+SAHA+2        0 -0.575830  0.614180  0.676500  0.127380   
292              SAHA+SAHA+3        0 -0.039234  0.357270  0.467940  0.189710   
294              SAHA+SAHA+4        0  0.042073  0.090531  0.140260  0.093571   
296              SAHA+SAHA+5        0 -0.037934 -0.052660 -0.028860  0.057298   
298              SAHA+SAHA+6        0 -0.025114 -0.112470 -0.111420 -0.072167   

          ks5       ks6       ks7       ks8  ...     ks229     ks230  \
12  -0.887700 -0.625570 -0.599290 -0.572840  ... -0.185950  0.197050   
14  -0.644780 -0.131510  0.334720  0.418330  ...  0.173700  0.161250   
16   0.195020 -0.093485 -0.094498 -0.089402  ...  0.484000  0.292200   
18   0.155430  0.110520  0.096807  0.088164  ...  0.407050  0.313020   
20  -0.222490 -0.169770 -0.157740 -0.151010  ...  0.036541  0.060248   
..        ...       ...       ...       ...  ...       ...       ...   
290  0.676570  0.688240  0.657050  0.620200  ...  0.501140  0.155110   
292  0.540580  0.464200  0.437200  0.413260  ...  0.075193  0.030012   
294  0.183460  0.141350  0.129490  0.116210  ... -0.063889  0.024933   
296  0.026528 -0.030210 -0.030452 -0.033717  ... -0.067626  0.041535   
298 -0.133120 -0.117890 -0.103570 -0.097284  ... -0.070852 -0.022251   

        ks231     ks232     ks233     ks234     ks235     ks236     ks237  \
12  -0.246260 -0.134130 -0.121170  0.197050 -0.292570 -0.349690 -0.322820   
14  -0.391330  0.255540  0.227910  0.161250 -0.654050 -0.670060 -0.371160   
16  -0.367780  0.699280  0.635780  0.292200 -0.463030 -0.545270 -0.203230   
18  -0.230140  0.620040  0.570010  0.313020 -0.455040 -0.469060 -0.143330   
20  -0.121030  0.080550  0.058614  0.060248 -0.137510 -0.156480 -0.074083   
..        ...       ...       ...       ...       ...       ...       ...   
290 -0.155500  0.597550  0.585120  0.155110 -0.497830 -0.492170 -0.133120   
292  0.097150  0.078057  0.084993  0.030012  0.165220  0.154520 -0.039340   
294  0.056691 -0.061606 -0.064976  0.024933  0.112500  0.081841 -0.028175   
296 -0.058015 -0.054617 -0.068953  0.041535  0.077927 -0.056419 -0.059348   
298 -0.058545 -0.071344 -0.080024 -0.022251  0.045747 -0.036961 -0.051677   

        ks238  
12   0.267480  
14   0.469350  
16   0.417000  
18   0.260370  
20   0.104870  
..        ...  
290  0.241290  
292 -0.085260  
294 -0.053385  
296  0.047265  
298  0.049158  

[120 rows x 240 columns]

In [182]:
#lda model
#just sample
#Dna ： etoposide，Gemcitabine
#mt：paclitaxel，Vinblastine
#hsp：17aag，auy922

#prot："MG132"，ps341
#hdac：saha，apicidin
from sklearn.datasets import make_classification
from sklearn.discriminant_analysis import LinearDiscriminantAnalysis

X_drug=training.iloc[:, 3:240].values

y=[]
for i in range(24):
    y.append('DNA')
for j in range(24):
    y.append('MT')
for k in range(24):
    y.append('HSP')
for m in range(24):
    y.append('PROT')
for n in range(24):
    y.append('HDAC')
print(len(X_drug[0]))
model = LinearDiscriminantAnalysis(n_components=4)
#model.fit(X_drug, y)
#X_drug2 = model.transform(X_drug)
#reduce dimension for the data
X_drug2=model.fit(X_drug,y).transform(X_drug)
X_drug2
print(len(X_drug2[0]))



237
4


In [196]:
#pca_drug = PCA(n_components=6)
#components = pca_drug.fit_transform(X_drug)
principalDf = pd.DataFrame(data = X_drug2
             , columns =['principal component 1', 'principal component 2','principal component 3','principal component 4'])
principalDf

principal component 1  principal component 2  principal component 3  \
0                -3.252321              -0.549709               4.233718   
1                -3.428148              -0.461099               4.660474   
2                -3.631864              -0.755152               4.348255   
3                -3.405172              -0.528072               4.279064   
4                -3.572289               1.157390               3.429835   
..                     ...                    ...                    ...   
115               1.453716               7.831404               0.252974   
116               0.983922               5.738939               1.050120   
117               2.602001               6.308003               1.520111   
118               1.551104               7.264680               0.095375   
119               2.206484               7.439078               0.757224   

     principal component 4  
0                 2.384586  
1                 1.632272  
2                 1.386191  
3                 1.937831  
4                 0.583612  
..                     ...  
115               0.863162  
116              -2.065507  
117               0.106051  
118              -0.003251  
119               0.679434  

[120 rows x 4 columns]

In [183]:
#predict
#still sample but useful this block 
#model2 = LinearDiscriminantAnalysis(n_components=1)
#model2.fit(X_drug2,y)
#row=X_drug2
#yhat = model2.predict(row)
# summarize prediction
#print('Predicted Class: %s' % yhat)



In [184]:
all_list=[]

for i in range(len(X_drug2)):
    all_list.append(X_drug2[i])
print(len(all_list))

120


In [192]:
X_training=[]
X_testing=[]

X_training=all_list[6:24]+all_list[30:48]+all_list[54:72]+all_list[78:96]+all_list[102:120]

X_testing=all_list[:6]+all_list[24:30]+all_list[48:54]+all_list[72:78]+all_list[96:102]
#X_testing.append()
print(len(X_training))
print(len(X_testing))
y=[]
for i in range(18):
    y.append('DNA')
for j in range(18):
    y.append("MT")
for k in range(18):
    y.append("HSP")
for m in range(18):
    y.append("PROT")
for n in range(18):
    y.append("HDAC")
model2 = LinearDiscriminantAnalysis(n_components=4)
model2.fit(X_training,y)
yhat2=model2.predict(X_testing)

print(yhat2)

90
30
['DNA' 'DNA' 'DNA' 'DNA' 'DNA' 'DNA' 'MT' 'MT' 'MT' 'MT' 'MT' 'MT' 'HSP'
 'HSP' 'HSP' 'HSP' 'HSP' 'HSP' 'PROT' 'PROT' 'PROT' 'PROT' 'PROT' 'PROT'
 'HDAC' 'HDAC' 'HDAC' 'HDAC' 'HDAC' 'HDAC']


In [193]:
def accuracy(list1,list2):
    length=len(list1)
    list3=[]
    for i in range(int(length/5.0)):
        list3.append(list2[0])
    for j in range(int(length/5.0)):
        list3.append(list2[1])
    for m in range(int(length/5.0)):
        list3.append(list2[2])
    for n in range(int(length/5.0)):
        list3.append(list2[3])
    for k in range(int(length/5.0)):
        list3.append(list2[4])

    counter=0
    for k in range(len(list1)):

        if list1[k]==list3[k]:
            counter+=1
    return counter/length



In [194]:
#test accuracy
'%.2f%%' % ((accuracy( yhat2,['DNA',"MT","HSP","PROT","HDAC"]))*100)

'100.00%'

In [217]:
import plotly.express as px
from sklearn.decomposition import PCA
from sklearn.preprocessing import StandardScaler
model = LinearDiscriminantAnalysis(n_components=4)
y2=[]
for i in range(24):
    y2.append('DNA')
for j in range(24):
    y2.append('MT')
for k in range(24):
    y2.append('HSP')
for m in range(24):
    y2.append('PROT')
for n in range(24):
    y2.append('HDAC')
#model.fit(X_drug, y)
#X_drug2 = model.transform(X_drug)
X_drug3=training.iloc[:,2:240].values
X_drug3=model.fit(X_drug3,y2).transform(X_drug3)
fig = px.scatter(X_drug3, x=0, y=1, color=training['drug_name'])
fig.show()
#these 5 groups are very clearly separated, as we can see in the image

In [171]:
# Create a function
def select_n_components(var_ratio, goal_var: float) -> int:
    # Set initial variance explained so far
    total_variance = 0.0
    
    # Set initial number of features
    n_components = 0
    
    # For the explained variance of each feature:
    for explained_variance in var_ratio:
        
        # Add the explained variance to the total
        total_variance += explained_variance
        
        # Add one to the number of components
        n_components += 1
        
        # If we reach our goal level of explained variance
        if total_variance >= goal_var:
            # End the loop
            break
            
    # Return the number of components
    return n_components

In [173]:
lda = LinearDiscriminantAnalysis(n_components=None)
y5=[]
for i in range(18):
    y5.append('DNA')
for j in range(18):
    y5.append("MT")
for k in range(18):
    y5.append("HSP")
for m in range(18):
    y5.append("PROT")
for n in range(18):
    y5.append("HDAC")

X_lda=lda.fit(X_training,y5)


In [178]:
# Create array of explained variance ratios
lda_var_ratios = lda.explained_variance_ratio_
lda_var_ratios

array([0.4150517 , 0.33917834, 0.17749731, 0.06827265])

In [180]:
# Run function
print(select_n_components(lda_var_ratios, 0.95))
print(select_n_components(lda_var_ratios, 0.85))
print(select_n_components(lda_var_ratios, 0.75))
print(select_n_components(lda_var_ratios, 0.65))
print(select_n_components(lda_var_ratios, 0.35))


4
3
2
2
1


In [197]:
#pca_drug = PCA(n_components=6)
#components = pca_drug.fit_transform(X_drug)
principalDf = pd.DataFrame(data = X_drug2
             , columns =['principal component 1', 'principal component 2','principal component 3','principal component 4'])
principalDf

principal component 1  principal component 2  principal component 3  \
0                -3.252321              -0.549709               4.233718   
1                -3.428148              -0.461099               4.660474   
2                -3.631864              -0.755152               4.348255   
3                -3.405172              -0.528072               4.279064   
4                -3.572289               1.157390               3.429835   
..                     ...                    ...                    ...   
115               1.453716               7.831404               0.252974   
116               0.983922               5.738939               1.050120   
117               2.602001               6.308003               1.520111   
118               1.551104               7.264680               0.095375   
119               2.206484               7.439078               0.757224   

     principal component 4  
0                 2.384586  
1                 1.632272  
2                 1.386191  
3                 1.937831  
4                 0.583612  
..                     ...  
115               0.863162  
116              -2.065507  
117               0.106051  
118              -0.003251  
119               0.679434  

[120 rows x 4 columns]

In [200]:
training

drug_name  default       ks1       ks2       ks3       ks4  \
12   Etoposide+Gemcitabine+1        0 -0.825000 -0.463430 -0.623680 -0.811710   
14   Etoposide+Gemcitabine+2        0 -0.850830 -0.166270  0.283220 -0.518290   
16   Etoposide+Gemcitabine+3        0 -0.690000  0.233580 -0.093266  0.163760   
18   Etoposide+Gemcitabine+4        0 -0.682500  0.227200  0.109890  0.123580   
20   Etoposide+Gemcitabine+5        0 -0.153740 -0.136810 -0.168360 -0.151440   
..                       ...      ...       ...       ...       ...       ...   
290              SAHA+SAHA+2        0 -0.575830  0.614180  0.676500  0.127380   
292              SAHA+SAHA+3        0 -0.039234  0.357270  0.467940  0.189710   
294              SAHA+SAHA+4        0  0.042073  0.090531  0.140260  0.093571   
296              SAHA+SAHA+5        0 -0.037934 -0.052660 -0.028860  0.057298   
298              SAHA+SAHA+6        0 -0.025114 -0.112470 -0.111420 -0.072167   

          ks5       ks6       ks7       ks8  ...     ks229     ks230  \
12  -0.887700 -0.625570 -0.599290 -0.572840  ... -0.185950  0.197050   
14  -0.644780 -0.131510  0.334720  0.418330  ...  0.173700  0.161250   
16   0.195020 -0.093485 -0.094498 -0.089402  ...  0.484000  0.292200   
18   0.155430  0.110520  0.096807  0.088164  ...  0.407050  0.313020   
20  -0.222490 -0.169770 -0.157740 -0.151010  ...  0.036541  0.060248   
..        ...       ...       ...       ...  ...       ...       ...   
290  0.676570  0.688240  0.657050  0.620200  ...  0.501140  0.155110   
292  0.540580  0.464200  0.437200  0.413260  ...  0.075193  0.030012   
294  0.183460  0.141350  0.129490  0.116210  ... -0.063889  0.024933   
296  0.026528 -0.030210 -0.030452 -0.033717  ... -0.067626  0.041535   
298 -0.133120 -0.117890 -0.103570 -0.097284  ... -0.070852 -0.022251   

        ks231     ks232     ks233     ks234     ks235     ks236     ks237  \
12  -0.246260 -0.134130 -0.121170  0.197050 -0.292570 -0.349690 -0.322820   
14  -0.391330  0.255540  0.227910  0.161250 -0.654050 -0.670060 -0.371160   
16  -0.367780  0.699280  0.635780  0.292200 -0.463030 -0.545270 -0.203230   
18  -0.230140  0.620040  0.570010  0.313020 -0.455040 -0.469060 -0.143330   
20  -0.121030  0.080550  0.058614  0.060248 -0.137510 -0.156480 -0.074083   
..        ...       ...       ...       ...       ...       ...       ...   
290 -0.155500  0.597550  0.585120  0.155110 -0.497830 -0.492170 -0.133120   
292  0.097150  0.078057  0.084993  0.030012  0.165220  0.154520 -0.039340   
294  0.056691 -0.061606 -0.064976  0.024933  0.112500  0.081841 -0.028175   
296 -0.058015 -0.054617 -0.068953  0.041535  0.077927 -0.056419 -0.059348   
298 -0.058545 -0.071344 -0.080024 -0.022251  0.045747 -0.036961 -0.051677   

        ks238  
12   0.267480  
14   0.469350  
16   0.417000  
18   0.260370  
20   0.104870  
..        ...  
290  0.241290  
292 -0.085260  
294 -0.053385  
296  0.047265  
298  0.049158  

[120 rows x 240 columns]

In [216]:
#knn
from sklearn.neighbors import KNeighborsClassifier
from sklearn.metrics import accuracy_score
knn5 = KNeighborsClassifier(n_neighbors=10)
knn5.fit(X_training, y)
y_pred_5 = knn5.predict(X_testing)
print(y_pred_5)
print(accuracy(y_pred_5,['DNA',"MT","HSP","PROT","HDAC"]))
#print("Accuracy with k=5", accuracy_score(y_test, y_pred_5)*100)

['DNA' 'DNA' 'DNA' 'DNA' 'DNA' 'DNA' 'MT' 'MT' 'MT' 'MT' 'MT' 'MT' 'HSP'
 'HSP' 'HSP' 'HSP' 'HSP' 'HSP' 'PROT' 'PROT' 'PROT' 'PROT' 'PROT' 'PROT'
 'HDAC' 'HDAC' 'HDAC' 'HDAC' 'HDAC' 'HDAC']
1.0


## most results show the accuracy is 100%.
ks5: 93.3%
ks4: 100%
ks3: 100%
ks2: 100%
Maybe it is because our data is little. So the accuracy is incredibly high. 
And it is obvious that different classes are different. 

1.0